discovery-provider/src/queries/query_helpers.py

```diff
@@ -195,6 +195,23 @@ def populate_user_metadata(session, user_ids, users, current_user_id):
     )
     repost_count_dict = {user_id: repost_count for (user_id, repost_count) in repost_counts}
 
+    # build dict of user id --> repost count
+    track_save_counts = (
+        session.query(
+            Save.user_id,
+            func.count(Save.user_id)
+        )
+        .filter(
+            Save.is_current == True,
+            Save.is_delete == False,
+            Save.save_type == SaveType.track,
+            Save.user_id.in_(user_ids)
+        )
+        .group_by(Save.user_id)
+        .all()
+    )
+    track_save_count_dict = {user_id: user_track_save_count for (user_id, user_track_save_count) in track_save_counts}
```

Comment:
perhaps we should add a flag for this since we don't expose this information anywhere else in the UI and it will slow down all user fetches. wdyt?

Reply:
Do we want to build out a flagging system?
I feel like we've typically done all or nothing.

Reply:
ya, i can go either way. it's probably fine to include. defer to @SidSethi  here

Reply:
ye agreed with ray - let's make this flag and add comment to eventually move other fields to this structure as well